TCC.Lib/Benchmark/BenchmarkHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using TCC.Lib.Blocks;
+using TCC.Lib.Options;
+
+namespace TCC.Lib.Benchmark
+{
+    public static class BenchmarkHelper
+    {
+        private static IEnumerable<BenchmarIteration> GenerateBenchmarkIteration(this BenchmarkOption benchmarkOption)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this method into two, one handling parameters check and the other handling the iterator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS4456)